File: apps/api/internal/model/top_inst.go
package model

import (
	"time"
)

// TopInst 龙虎榜机构交易名单模型
type TopInst struct {
	ID        uint      `json:"id" gorm:"primaryKey"`
	TradeDate string    `json:"trade_date" gorm:"size:8;index;index:idx_code_date_exalter"` // 交易日期 YYYYMMDD
	TSCode    string    `json:"ts_code" gorm:"size:20;index;index:idx_code_date_exalter"`   // 股票代码
	Exalter   string    `json:"exalter" gorm:"size:200;index;index:idx_code_date_exalter"`  // 营业部名称
	Buy       float64   `json:"buy"`                                                          // 买入额(万)
	BuyRate   float64   `json:"buy_rate"`                                                     // 买入占总成交比例
	Sell      float64   `json:"sell"`                                                         // 卖出额(万)
	SellRate  float64   `json:"sell_rate"`                                                    // 卖出占总成交比例
	NetBuy    float64   `json:"net_buy"`                                                      // 净买额(万)
	Side      string    `json:"side" gorm:"size:20"`                                          // 买卖方向
	Reason    string    `json:"reason" gorm:"size:500"`                                       // 上榜原因
	Source    string    `json:"source" gorm:"size:50;index"`                                  // 数据来源
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

func (TopInst) TableName() string {
	return "top_inst"
}

// ToResponse 转换为响应格式
func (t *TopInst) ToResponse() map[string]interface{} {
	return map[string]interface{}{
		"id":         t.ID,
		"trade_date": formatDate(t.TradeDate),
		"ts_code":    t.TSCode,
		"exalter":    t.Exalter,
		"buy":        t.Buy,
		"buy_rate":   t.BuyRate,
		"sell":       t.Sell,
		"sell_rate":  t.SellRate,
		"net_buy":    t.NetBuy,
		"side":       t.Side,
		"reason":     t.Reason,
		"source":     t.Source,
	}
}

File: apps/web-admin/src/services/topinst.ts
/**
 * 龙虎榜机构交易名单 API 服务
 */
import api from './api'

export interface TopInstItem {
  id: string
  trade_date: string
  ts_code: string
  exalter: string
  buy: number
  buy_rate: number
  sell: number
  sell_rate: number
  net_buy: number
  side: string
  reason: string
  source: string
}

export interface TopInstQueryParams {
  page?: number
  pageSize?: number
  ts_code?: string
  exalter?: string
  startDate?: string
  endDate?: string
  min_buy?: number
  max_buy?: number
}

export interface TopInstResponse {
  data: TopInstItem[]
  page: number
  pageSize: number
  total: number
}

export const topInstApi = {
  /**
   * 获取龙虎榜机构交易名单列表
   */
  getTopInstList: async (params: TopInstQueryParams): Promise<TopInstResponse> => {
    const { data } = await api.get('/v1/admin/datasource/top-inst', { params })
    return data
  },

  /**
   * 删除单条龙虎榜机构交易名单数据
   */
  deleteTopInst: async (id: string) => {
    const { data } = await api.delete(`/v1/admin/datasource/top-inst/${id}`)
    return data
  },

  /**
   * 批量删除龙虎榜机构交易名单数据
   */
  batchDeleteTopInst: async (ids: string[]) => {
    const { data } = await api.post('/v1/admin/datasource/top-inst/batch-delete', { ids })
    return data
  },

  /**
   * 同步龙虎榜机构交易名单数据（手动触发）
   */
  syncTopInst: async (tradeDate?: string) => {
    const params = tradeDate ? { trade_date: tradeDate } : {}
    const { data } = await api.post('/v1/admin/datasource/top-inst/sync', null, { params })
    return data
  },

  /**
   * 获取龙虎榜机构交易名单统计
   */
  getStats: async () => {
    const { data } = await api.get('/v1/admin/datasource/top-inst/stats')
    return data
  },

  /**
   * 获取龙虎榜机构交易名单同步设置
   */
  getSyncSettings: async () => {
    const { data } = await api.get('/v1/admin/settings/top-inst-sync')
    return data
  },

  /**
   * 保存龙虎榜机构交易名单同步设置
   */
  saveSyncSettings: async (settings: { enabled: boolean; fixed_times: string[] }) => {
    const { data } = await api.post('/v1/admin/settings/top-inst-sync', settings)
    return data
  },
}
